main.go

```diff
@@ -133,6 +137,12 @@ func run() error {
 	reconciler.DNSProvider = f.Provider()
 	log.Info().Str("provider", reconciler.DNSProvider.String()).Msg("Started")
 
+	err = inferClusterGeoTag(config, mgr.GetAPIReader())
+	if err != nil {
+		log.Err(err).Msg("Can't infer the CLUSTER_GEO_TAG")
```

Comment:
In order to get as many input values as possible as soon as possible (and fail on invalid inputs soon as possible), would it be possible  to make the initialization (`inferClusterGeotag`) as on the top of `main()` as possible? Maybe  somewhere after the manager initialization.

Reply:
done, well calling from main now, but sooner